Sources/Protocols/CellItem/CollectionViewCellItem.swift

```diff
@@ -0,0 +1,324 @@
+                                        CollectionViewGeneralCellItem,
+                                        CollectionViewCellItemDataSource,
+                                        CollectionViewSiblingCellItem {
+    
+}
+
+// MARK: - CollectionViewReuseCellItem
+
+public protocol CollectionViewReuseCellItem: AnyObject {
+    var reuseType: ReuseType { get }
+}
+
+// MARK: - CollectionViewSizeCellItem
+
+public protocol CollectionViewSizeCellItem: AnyObject {
+    func size() -> CGSize
+}
+
+// MARK: - CollectionViewConfigureCellItem
+
+public protocol CollectionViewConfigureCellItem: AnyObject {
+    func configure(_ cell: UICollectionViewCell)
+}
+
+// MARK: - CollectionViewSiblingCellItem
+
+public protocol CollectionViewSiblingCellItem: AnyObject {
+    var collectionView: UICollectionView { get set }
+    var indexPath: IndexPath { get set }
+    var sectionItem: CollectionViewSectionItem? { get set }
+}
+
+extension CollectionViewSiblingCellItem {
+    public var collectionView: UICollectionView {
+        get {
+            if let object = objc_getAssociatedObject(self, &AssociatedKeys.collectionView) as? UICollectionView {
+                return object
+            }
+            fatalError("You should never get this error if you use collection view tools properly. " +
+                               "The reason is that you create cell item and didn't set collection view")
+        }
+        set {
+            objc_setAssociatedObject(self, &AssociatedKeys.collectionView, newValue, .OBJC_ASSOCIATION_ASSIGN)
+        }
+    }
+    
+    public var indexPath: IndexPath {
+        get {
+            if let object = objc_getAssociatedObject(self, &AssociatedKeys.indexPath) as? IndexPath {
+                return object
+            }
+            fatalError("You should never get this error if you use collection view tools properly. " +
+                               "The reason is that you create cell item and didn't set index path")
+        }
+        set {
+            objc_setAssociatedObject(self, &AssociatedKeys.indexPath, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+    
+    public weak var sectionItem: CollectionViewSectionItem? {
+        get {
+            if let object = objc_getAssociatedObject(self, &AssociatedKeys.sectionItem) as? CollectionViewSectionItem {
+                return object
+            }
+            return nil
+        }
+        set {
+            objc_setAssociatedObject(self, &AssociatedKeys.sectionItem, newValue, .OBJC_ASSOCIATION_ASSIGN)
+        }
+    }
+}
+
+// MARK: - CollectionViewGeneralCellItem
+
+public typealias ActionHandler = () -> Void
+public typealias ActionResolver = () -> Bool
+public typealias CellActionHandler = (UICollectionViewCell) -> Void
+public typealias ViewActionHandler = (UICollectionReusableView, String, UICollectionView, IndexPath) -> Void
+
+public protocol CollectionViewGeneralCellItem: AnyObject {
```

Comment:
Can you remind me why we declare it as protocol at all? 
It conforms to `AnyObject` which basically means cell item could only be a class anyway, it requires a ton of associated properties and uses default implementations in extension, which are quite finicky when it comes to subclassing (https://bugs.swift.org/browse/SR-103).

Reply:
All of these protocols weren't conform to `AnyObject` in the beginning. It caused "change `let` to `var`" behaviour after setting any handler and there are only one solution to it - add `AnyObject` conformance.
Anyway I believe that protocols make cell items more flexible. There are no needs to inherit from some class and any object can become `CellItem`.
@artemnovichkov, @Clanat, @medvedzzz let's discuss it and decide what to do?